mycompany-seeq-link-connector-myconnector/src/main/java/com/mycompany/seeq/link/connector/MyConnection.java

```diff
@@ -142,87 +170,95 @@ public void disconnect() {
-                    // iteration is short-circuited for any reason.
-                });
-    }
+        Stream.Builder<Sample> streamBuilder = Stream.builder();
 
-    @Override
-    public Integer getMaxConcurrentRequests() {
-        // This parameter can help control the load that Seeq puts on an external datasource. It is typically
-        // controlled from the configuration file.
-        return this.connectionConfig.getMaxConcurrentRequests();
+        for (DatasourceSimulator.Tag.Value tagValue: tagValues) {
+            streamBuilder.accept(new Sample(tagValue.getTimestamp(), tagValue.getMeasure()));
+        }
+
+        return streamBuilder.build();
     }
 
     @Override
-    public Integer getMaxResultsPerRequest() {
-        // This parameter can help control the load and memory usage that Seeq puts on an external datasource. It is
-        // typically controlled from the configuration file.
-        return this.connectionConfig.getMaxResultsPerRequest();
+    public Stream<Capsule> getCapsules(GetCapsulesParameters parameters) throws Exception {
+        try {
+            // This is an example of how you may query your datasource for alarm events and is specific to the
+            // simulator example. This should be replaced with your own datasource-specific call.
+            Iterable<DatasourceSimulator.Alarm.Event> events = this.datasourceSimulator.getAlarmEvents(
+                    parameters.getDataId(),
+                    parameters.getStartTime(),
+                    parameters.getEndTime(),
+                    parameters.getCapsuleLimit()
+            );
+
+            // Return a Stream to iterate through all the capsules in the time range.
+            //
+            // Streams are important to use here to avoid bringing all the data into memory to satisfy the
+            // request. The Seeq connector host will automatically "page" the data upload so that we don't hit memory
+            // ceilings on large requests. Streams can be created in a variety of ways, such as Guava's
+            // Streams.stream(iterable), Java's Stream.of(T... values), or Collection.stream().
+            //
+            // The code within this function is largely specific to the simulator example. But it should give you an
+            // idea of
+            // some of the concerns you'll need to attend to.
+            Stream.Builder<Capsule> streamBuilder = Stream.builder();
+
+            for (DatasourceSimulator.Alarm.Event event : events) {
+                TimeInstant start = new TimeInstant(event.getStart());
+                TimeInstant end = new TimeInstant(event.getEnd());
+
+                List<Capsule.Property> capsuleProperties = new ArrayList<>();
+                capsuleProperties.add(new Capsule.Property("Value", Double.toString(event.getIntensity()), "rads"));
+
+                streamBuilder.accept(new Capsule(start, end, capsuleProperties));
+            }
+
+            return streamBuilder.build();
```

Comment:
I think similarly to the `tagValues` comment it would be more natural to avoid a `Stream.Builder` here and instead use stream operations like `map` to transform the iterable into a stream.